### PR TITLE
Add readme file for this repo to detail setup instructions

### PR DESCRIPTION
**Pull request by WPprodigy**
As promised, here's an initial attempt at a readme file that helps outline the setup process for a test site. At the moment, it's pretty tough to start from scratch and figure out all the pieces.

I'm thinking the easiest way to get people started quickly is to have an XML import file. I've put together a pretty clean one, with just the minimal pages and menu items needed to get going. Not sure where we should host this though, any ideas? Need to update the `[import this xml file](#TODO)` part in the readme once we have it live.

Lastly, I'm thinking some sort generator with faker data would be useful for creating a decent/bulk amount of pledges. Ran out of time, but can look into this later.
_____

Can view the readme here with the GH-processed markup: https://github.com/WPprodigy/five-for-the-future/tree/add/readme

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+## Contributing
+
+In order to contribute with code changes, you'll want to set up a local environment to test changes and then push the changes to a Pull Request on this Github Repository.
+
+### Initial environment setup
+
+1) Use whichever local WordPress development setup you prefer and create a new local WP site.
+2) Find the `wp-content` folder and delete it (make a backup if you have data you don't want to lose data you already have there).
+3) Fork the [five-for-the-future](https://github.com/WordPress/five-for-the-future) repository under your own Github account.
+4) Run `git clone git@github.com:WordPress/five-for-the-future.git wp-content`. Replace `WordPress/` with your Github `USERNAME/` to clone your newly forked repo instead.
```
Comment (ryelle) [inline readme.md]: Maybe this, so that folks who just copy-paste don't accidentally clone the wrong repo?
```suggestion
4) Run `git clone git@github.com:[your username]/five-for-the-future.git wp-content`, replacing `[your username]` with your github username to clone your forked repo.
```

File: readme.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+5) Ensure this newly cloned `wp-content` folder is where it should be in the WP structure.
+6) Copy over the base theme with: `svn export https://meta.svn.wordpress.org/sites/trunk/wordpress.org/public_html/wp-content/themes/pub/wporg themes/pub/wporg` (this should be run from the `wp-content` folder).
+
+### Configuring the site
+
+1) Login to your site and activate the "Five for the Future" theme and plugin.
+2) Navigate to `wp-content/themes/pub/wporg` from your CLI and run: `npm install && grunt default`
```
Comment (ryelle) [inline readme.md]: Do we need this step? (I don't remember doing it, but I set things up differently)

File: readme.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+If you making changes to the theme's CSS, you can run `npm start` at `/wp-content/themes/wporg-5ftf` to watch for CSS changes and automatically compile.
+
+If you are making changes to the plugins, you can run `composer install` at `/wp-content/plugins/wporg-5ftf` and then `composer run test` to run the WP unit tests.
+
+And lastly, you can run PHPCS for both the theme and the plugin at the root `/wp-content/` folder by running `composer install` there once, followed by `composer run phpcs` when you want to code scan. 
+
+### Submitting Pull Requests
+
+The first thing you'll want to do before changing any code is create a new branch based on the production/master branch. Then you can commit your code changes locally and push this new branch to your forked repository on Github. Then visit the [official repository](https://github.com/WordPress/five-for-the-future/) and you should see the option to open up a Pull Request based on the recently pushed branch on your fork.
```
Comment (ryelle) [inline readme.md]: Let's just call the branch `production`, since that is the name of the branch itself, and using "production/master" might be confusing if folks try to look for `master`. So like "… create a new branch based on the `production` branch."

**Review by ryelle — COMMENTED** [review]
Thanks for writing this up! I left a few small comments, but overall looks good 😁

**Comment by ryelle** [comment]
> XML import file. […] Not sure where we should host this though, any ideas?

Hm... Can we upload xml to the make sites? Or maybe a github gist?

> Lastly, I'm thinking some sort generator with faker data would be useful for creating a decent/bulk amount of pledges. Ran out of time, but can look into this later

This is an interesting idea… I think you get better test case scenarios when using "real" data, but generated data can show unexpected edge cases. At least, that was my experience with [wc-smooth-generator](https://github.com/woocommerce/wc-smooth-generator) – interesting edge cases that would rarely happen in real life. If you want to build this, cool - but a basic import file would be used more, I think.

File: readme.md (hunk 2 above)
Comment (WPprodigy) [reply readme.md]: Oh, guess not! The parent theme uses `get_theme_file_uri()` to enqueue `css/style.css` so there is indeed no need to build it in the parent since only the child's is used.

**Comment by WPprodigy** [comment]
Here's the XML export I put together: https://gist.github.com/WPprodigy/4f5f919a7fab0866ea9c28d202becd33. Should we put it under the `WordPress` gist account?

And yeah, I agree that adding some pledges to this would be ideal / easiest for most to use. Can add to the export at some point in the future after I've got a good way to bulk generate a reasonable number of them.

**Comment by iandunn** [comment]
What about moving adding the file as something like `/data/default-content.xml` in the repo itself?

IIRC, Make sites don't let you upload XML, and none of the contributors here have admin rights on the `WordPress` GH account. We could ask someone to add it, but that'd introduce unnecessary friction when we want to update it in the future.

**Comment by iandunn** [comment]
I merged this into the readme that was created after this PR, and added the WXR in the `/.env` folder. Thanks!
